src/audiovideoobserver/AudioVideoObserver.ts

```diff
@@ -97,9 +97,8 @@ export default interface AudioVideoObserver {
    * * `MeetingSessionStatusCode.OK`: `demoteFromPrimaryMeeting` was used to remove the attendee.
    * * `MeetingSessionStatusCode.AudioAuthenticationRejected`: `chime::DeleteAttendee` was called on the Primary
    *   meeting attendee used in `promoteToPrimaryMeeting`.
-   * * `MeetingSessionStatusCode.SignalingBadRequest`: Other failure, possibly due to disconnect
-   *   or timeout. These failures are likely retryable. Any disconnection will trigger an automatic
-   *   demotion to avoid unexpected or unwanted promotion state on reconnection.
+   * * `MeetingSessionStatusCode.AudioVideoDisconnectedWhilePromoted`: The client disconnected (and may be reconnecting).
```

Comment:
I am not sure if we maintain a posture that a reconnect is idempotent and currently when a reconnect happens a builder needs to not do anything. Here, it seems we are deviating from that path no hard requirement. Is there a way to fix the promotion/demotion and not disconnect on the reconnect and upon reconnect the attendee gets promoted back up, or this is complex and not needed from UX?

Reply:
As mentioned above, this is behavior that was intentionally decided on when this feature released, which is why its documented (this comment is just refactored). On the backend, we intentionally do not keep state for connections that do not exist, which is why this would be very complicated to change.  Additionally, this is a safety mechanisms for builders and the end users; you don't have attendees randomly popping back onto the stage when they disconnect, if it takes them a while to reconnect and you move on.

Reply:
The latter is tbh the more important point. Think of it as the backend intentionally demoting the attendee if they disconnect.

Reply:
Got it, thanks, I missed reading the above earlier documentation.